GitHubCodespaces.ps1

```diff
@@ -0,0 +1,538 @@
+    }
+
+    $uriFragment = [String]::Empty
+    $description = [String]::Empty
+    switch ($PSCmdlet.ParameterSetName)
+    {
+        'AuthenticatedUser' {
+            # /user/codespaces
+            $uriFragment = 'user/codespaces'
+            $description = 'Getting codespaces for current authenticated user'
+
+            break
+        }
+
+        'CodespaceName' {
+            $telemetryProperties['CodespaceName'] = Get-PiiSafeString -PlainText $CodespaceName
+
+            $uriFragment = "user/codespaces/$CodespaceName"
+            $description = "Getting user/codespaces/$CodespaceName"
+
+            break
+        }
+
+        'Organization' {
+            # /orgs/{org}/codespaces
+            # /orgs/{org}/members/{username}/codespaces
+
+            $telemetryProperties['OrganizationName'] = Get-PiiSafeString -PlainText $OrganizationName
+            if ([string]::IsNullOrWhiteSpace($UserName)) {
+                $uriFragment = "orgs/$OrganizationName/codespaces"
+                $description = "Getting codespaces for $OrganizationName"
+            }
+            else {
+                $telemetryProperties['UserName'] = Get-PiiSafeString -PlainText $UserName
+                $uriFragment = "orgs/$OrganizationName/members/$UserName/codespaces"
+                $description = "Getting codespaces for $OrganizationName"
+            }
+
+            break
+        }
+
+        'Repository' {
+            $elements = Resolve-RepositoryElements
+            $OwnerName = $elements.ownerName
+            $RepositoryName = $elements.repositoryName
+
+            $telemetryProperties['OwnerName'] = Get-PiiSafeString -PlainText $OwnerName
+            $telemetryProperties['RepositoryName'] = Get-PiiSafeString -PlainText $RepositoryName
+
+            $uriFragment = "repos/$OwnerName/$RepositoryName/codespaces"
+            $description = "Getting $OwnerName/$RepositoryName/codespaces"
+
+            break
+        }
+
+        'Uri' {
+            $elements = Resolve-RepositoryElements
+            $OwnerName = $elements.ownerName
+            $RepositoryName = $elements.repositoryName
+
+            $telemetryProperties['OwnerName'] = Get-PiiSafeString -PlainText $OwnerName
+            $telemetryProperties['RepositoryName'] = Get-PiiSafeString -PlainText $RepositoryName
+
+            $uriFragment = "repos/$OwnerName/$RepositoryName/codespaces"
+            $description = "Getting $OwnerName/$RepositoryName/codespaces"
+
+            break
+        }
+    }
+
+    $params = @{
+        'UriFragment' = $uriFragment
+        'Description' = $description
+        'AcceptHeader' = "$script:nebulaAcceptHeader,$script:baptisteAcceptHeader,$script:mercyAcceptHeader"
+        'AccessToken' = $AccessToken
+        'TelemetryEventName' = $MyInvocation.MyCommand.Name
+        'TelemetryProperties' = $telemetryProperties
+    }
+
+    $result = Invoke-GHRestMethodMultipleResult @params
+    if ($result.codespaces) {
```

Comment:
nit: Again, open brackets (`{`) on a separate line please.  Assume I say that everywhere else in this file as well.